### PR TITLE
add backward(is_train=False) and always mode for dropout

### PR DESCRIPTION
**Pull request by piiswrong**

### CODE REVIEW

**Comment by piiswrong** [comment]
@zackchase

**Comment by piiswrong** [comment]
@leezu This PR separates is_train and is_recording. There is a note in the doc of record() and pause(). But you can add it to other places if necessary

File: src/operator/dropout-inl.h [hunk 1]
```diff
@@ -96,7 +104,7 @@ class DropoutOp : public Operator {
       bernoulli_generate(count, this->pkeep_, maskptr);
   #pragma omp parallel for
       for (int i = 0; i < count; ++i) {
-        outptr[i] = dataptr[i] * maskptr[i];
+        outptr[i] = dataptr[i] * maskptr[i] * (1.0f / pkeep_);
```
Comment (piiswrong) [inline src/operator/dropout-inl.h]: @zhenlinluo @ykim362 
MKL dropout has been wrong the whole time!

Comment (ykim362) [inline src/operator/dropout-inl.h]: @piiswrong Got it, Thanks! I am looking into the code again to make sure other things are correct.

**Comment by leezu** [comment]
@piiswrong Thanks, this makes the is_train behaviour better documented. But the documentation is currently not shown anywhere on mxnet.io/api/python/ .. I tried adding it via #7327 , though please doublecheck the change as I can't build the docs locally.

**Comment by hassonofer** [comment]
@piiswrong is_train was added to the Executor at python/mxnet/executor.py but not at python/mxnet/module/module.py which is still `def backward(self, out_grads=None):`

Any specific reason why ?

**Pull request merged**